### ISSUE TITLE
--vo=gpu broken on Windows 10 after Patch Tuesday

### ISSUE THREAD

**Issue by nslay** [mpv-player/mpv#11007]
### Important Information

Provide following Information:
- mpv version
mpv 0.35.0-33-gcf349d68e3 Copyright © 2000-2022 mpv/MPlayer/mplayer2 projects
 built on Sun Dec 11 12:20:06 2022
libplacebo version: v5.229.1-33-g7ead30d
FFmpeg version: git-2022-12-10-07e0732a9
FFmpeg library versions:
   libavutil       57.43.100
   libavcodec      59.55.100
   libavformat     59.34.102
   libswscale      6.8.112
   libavfilter     8.52.100
   libswresample   4.9.100

- Windows Version
Windows 10 Home
10.0.19044 Build 19044

- Source of the mpv binary
updater.bat from mpv Windows release

- If known which version of mpv introduced the problem
I think Windows Update broke this. Everything worked before Windows Update ran last night. I've updated the GPU driver to the latest version (NVIDIA NVS 42000M).

- Possible screenshot or video of visual glitches
![mpvBrokenGPUVideoOutput](https://user-images.githubusercontent.com/6295986/207670239-f8cede7c-8784-46a5-862e-7a46af60eb44.png)

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps
Using --no-config or --vo=gpu, the mpv window is empty (though sound still plays, controls still work).
Workaround is to use --vo=direct3d.

Try to reproduce your issue with --no-config first. If it isn't reproducible
with --no-config try to first find out which option or script causes your issue.

Describe the reproduction steps as precise as possible. It's very likely that
the bug you experience wasn't reproduced by the developer because the workflow
differs from your own.
Run ```mpv --vo=gpu videoFile```
```gpu``` is the default video output when using --no-config.

### Expected behavior
Video should be visible in the mpv window.

### Actual behavior
The window is present and sized correctly but the window is an empty gray panel.

### Log file

Make a log file made with -v -v or --log-file=output.txt, paste it to
https://0x0.st/ or attach it to the github issue, and replace this text with a
link to it.

[mpv_output.txt](https://github.com/mpv-player/mpv/files/10230453/mpv_output.txt)

The issue will be closed for ignoring the issue template.

### Sample files

Sample files needed to reproduce this issue can be uploaded to https://0x0.st/
or similar sites. (Only needed if the issue cannot be reproduced without it.)
Do not use garbage like "cloud storage", especially not Google Drive.

You should be able to reproduce this with any ordinary video file. I've tried mp4, avi, webm and all have same problems when using --vo=gpu.

**Comment by CounterPillow** [comment]
> NOTE: Hey mpv developers. I really appreciate your work! But... maybe don't dump a user's folder contents into the log file when you expect us to post it on a completely open and public website!

Your secret is safe with me. 🫃

**Comment by Jules-A** [comment]
I can't reproduce this but I'm using an RX570 if that matters. I haven't had any issues with the latest Windows 10 update.

**Comment by nslay** [comment]
Everything works correctly on NVIDIA RTX3080 Ti, although I'm using the x64_64-v3 version of mpv for this different machine. Could be an NVIDIA sucks at old hardware problem.

**Comment by wbtcpip2** [comment]
i'm using libmpv 0.35.0-33-gcf349d68e3 win32 32bits with GTX 570 and video output is ok

**Comment by nslay** [comment]
Aha, I think I narrowed down the cause. Both mpv and VLC seem to be confused by the presence of 2 GPUs. There's a built-in Intel Graphics Card and then there is a discrete NVIDIA graphics card. Now the discrete NVIDIA graphics card has always been selected as the preferred graphics processing unit... since a decade ago when I picked this setting. When I did a clean GPU driver installation this morning, this setting was reset to "Auto Select" and then --vo=gpu produced video. I changed it to NVIDIA NVS 4200M and the video stopped working in the same fashion as I posted. So there's something with how mpv/VLC (but not video games or Netflix in firefox) works with the GPUs that was recently broken in a Patch Tuesday update. The latest driver on nvidia's site has a release date of 10/26/2021... it hasn't changed and was installed before Patch Tuesday when mpv functioned correctly. The only thing that changed is the Patch Tuesday update.

**Comment by Dudemanguy** [comment]
Is this still an issue?

**Comment by nslay** [comment]
Yep, just updated to today's build (updater.bat which got 20230430-git-6d422b3) and changed GPU in NVIDIA Control Panel from auto-select to high performance GPU (NVS 4400M) and can reproduce the issue still. If I change back to auto-select, everything works again. Mpv used to work with the NVIDIA GPU until some Windows patch Tuesday some months ago.

It seems to be related to having a built-in Intel GPU and a discrete NVIDIA GPU. Mpv otherwise works on machines with dedicated (though newer) NVIDIA GPUs on my end.
